deploy_tools/compile.py

```diff
@@ -0,0 +1,151 @@
+
+            if 'deployedBytecode' in evm_data:
+                result['deployedBytecode'] = add_0x_prefix(evm_data['deployedBytecode'].get('object', ''))
+
+        elif key == 'metadata':
+            if value:
+                result[key] = json.loads(value)
+
+        else:
+            result[key] = value
+
+    return result
+
+
+def normalize_compiled_contracts(compiled_contracts: Dict, file_paths: List[str]):
+    result: Dict[str, Dict] = {}
+
+    for source_path, file_contracts in compiled_contracts.items():
+        if source_path not in file_paths:
+            continue
+
+        for contract_name, raw_contract_data in file_contracts.items():
+            contract_data = normalize_contract_data(raw_contract_data)
+            if contract_name not in result:
+                result[contract_name] = contract_data
+            else:
+                raise BaseException('Can not compile two contracts with the same name')
+
+    return result
+
+
+def log_compilation_errors(errors: List[Dict]):
+    for error in errors:
+        if 'formattedMessage' in error:
+            print(error['formattedMessage'])
+        else:
+            print(error['message'])
+
+
+def compile_project(
+        contracts_path: str = None,
+        *,
+        file_paths: List[str] = None,
+        allow_paths: List[str] = None,
+        pattern='*.sol'
+):
+    """
+    Compiles all contracts of the project into a single output
+    Args:
+        contracts_path: The path of the folder that includes the contracts
+        file_paths: Optional a list of to compiled contracts can be provided
+        allow_paths: Additional paths from where it is allowed to load contracts
+        pattern: The pattern to find the solidity files
+
+    Returns: A dictionary containing the compiled assets of the contracts
+
+    """
+
+    if file_paths is None:
+        file_paths = []
+
+    if allow_paths is None:
+        allow_paths = []
+
+    if contracts_path is None and not file_paths:
+        contracts_path = 'contracts'
+
+    if contracts_path is not None:
+        file_paths.extend(find_files(contracts_path, pattern=pattern))
+        allow_paths.append(contracts_path)
+
+    sources = load_sources(file_paths)
+
+    std_input = {
+        'language': 'Solidity',
+        'sources': sources,
+        'settings': {
+            'outputSelection': {
+                '*': {
+                    '*': DEFAULT_OUTPUT_SELECTION
+                }
+            }
+        }
+    }
+
+    compilation_result = compile_standard(
+        std_input,
+        allow_paths=','.join(os.path.abspath(path) for path in allow_paths)
+    )
+
+    if 'errors' in compilation_result:
+        log_compilation_errors(compilation_result['errors'])
+
+    compiled_contracts = normalize_compiled_contracts(compilation_result['contracts'], file_paths)
+    return compiled_contracts
+
+
+def compile_contract(
+        name: str,
+        *,
+        contracts_path='contracts',
```

Comment:
What we want is the path to the contract we try to compile, not to multiple contracts. So I would change this to `contract_path`. Otherwise I would wonder whether the function is intended to compile one or multiple contracts when looking at the signature.

Reply:
Its not the path to the contract, its the path to all contracts and it will then search for the one that you want to compile. Not sure yet if that really makes sense.